haskell/security-advisories
# `cabal-audit`: init

## the cabal audit executable 

### usage 

```console
λ nix run github:mangoiv/security-advisories/mangoiv/cabal-audit-osv#cabal-audit --
trying to clone https://github.com/haskell/security-advisories
Cloning into '/tmp/cabal-audit-726d3e9345b766bc'...
remote: Enumerating objects: 172, done.
remote: Counting objects: 100% (172/172), done.
remote: Compressing objects: 100% (129/129), done.
remote: Total 172 (delta 6), reused 114 (delta 1), pack-reused 0
Receiving objects: 100% (172/172), 116.55 KiB | 1.31 MiB/s, done.
Resolving deltas: 100% (6/6), done.


Found advisories:

dependency "base" at version 4.18.1.0 is vulnerable for:
  HSEC-2023-0007 "readFloat: memory exhaustion with large exponent"
  published: 2024-04-23 12:43:30 +1000
  https://haskell.github.io/security-advisories/advisory/HSEC-2023-0007
  No fix version available
  toml, parser, dos

dependency "process" at version 1.6.17.0 is vulnerable for:
  HSEC-2024-0003 "process: command injection via argument list on Windows"
  published: 2024-04-23 12:43:30 +1000
  https://haskell.github.io/security-advisories/advisory/HSEC-2024-0003
  Fix available since version 1.6.19.0
  windows
```

> [!Note]
> if this causes some error wrt a lock file incompatibility, upgrade your nix version ;)

### features implemented 

- query the repository for vulnerabilities 
- provide local repository 
- pretty output 
- run the cabal solver on a project and match vulnerabilities against that 
- propose fix version
- link to security-advisories website 
- playtest

### open features 

- don't only provide the latest fix version but a range of fix versions
- check for the newest package on hackage that is greater than the fix version
- more configuration configure the output 
- json output instead of only pretty stdout 
- use API instead of repository
- more solver options (perhaps offer a special flags to pass on options to cabal)
- more structured monadic code (it's all IO for simplicity)

## other changes 

- gave the nix code a polish 
- gave the devshell a polish 
- introduced pre-commit-hooks.nix (please tell me if that's not wanted)
